### PR TITLE
BUG: tolerates !cite and fixes URI decoding

### LINKED ISSUE #87: Double check URI encoding between client and SW
There is something strange happening in https://github.com/qiime2/q2-diversity/issues/187 which causes q2view to act differently than `file://`.

### LINKED ISSUE #89: Allow custom `!cite` yaml type
Pass through for now until #86 is complete.

### PR DESCRIPTION
**Pull request by ebolyen**
fixes #89 
fixes #87

### CODE REVIEW

**Comment by ebolyen** [comment]
fixes https://github.com/qiime2/q2-diversity/issues/187

(Don't know if that will work across repos)

**Comment by thermokarst** [comment]
Doesn't look like it - the GH keyword highlight didn't light up the "fixes" word :(

File: app/lib/reader.js [hunk 1]
```diff
@@ -77,7 +77,8 @@ export default class Reader {
             }
             switch (event.data.type) {
             case 'GET_DATA':
-                this._getFile(event.data.filename).then((data) => {
+                // decode should go in the SW, but that'd require an upgrade
```
Comment (thermokarst) [inline app/lib/reader.js:80]: 👍

**Review by thermokarst — APPROVED** [review]
Tested locally, this works as advertised on both issues. Thanks!

**Pull request merged**